### PR TITLE
Enable configuration on disable anonymous user

### PR DESCRIPTION
**Pull request by cafeliker**
This is the proposed change on the issue #10760 

The administrator will be able to configure the Harbor instance to enable/disable the anonymous user with the UI like below:
image

Welcome for comments and feedback.

Thanks
ye

### CODE REVIEW

**Comment by reasonerjt** [comment]
IMO, it would be a better design if we add more level of control in project's public policy

**Comment by cafeliker** [comment]
> IMO, it would be a better design if we add more level of control in project's public policy

@reasonerjt Daniel, thanks for your comments, and can you be more specific?

ye

**Comment by cafeliker** [comment]
Add anonymous check in the search API

**Comment by cafeliker** [comment]
@steven-zou how will you merge this PR? I don't want to change every time you make a upgrade. Again, it should be the valid cyber security requirements if running Harbor in enterprise, specially when the enterprise Harbor service is internet facing.

**Comment by cafeliker** [comment]
@steven-zou I updated the codes to work with harbor v2.0.1, could you pls help review it again?

Thanks
ye

File: src/common/config/metadata/metadatalist.go [hunk 1]
```diff
@@ -121,6 +121,7 @@ var (
 
 		{Name: common.ProjectCreationRestriction, Scope: UserScope, Group: BasicGroup, EnvKey: "PROJECT_CREATION_RESTRICTION", DefaultValue: common.ProCrtRestrEveryone, ItemType: &ProjectCreationRestrictionType{}, Editable: false},
 		{Name: common.ReadOnly, Scope: UserScope, Group: BasicGroup, EnvKey: "READ_ONLY", DefaultValue: "false", ItemType: &BoolType{}, Editable: false},
+		{Name: common.DisableAnonymous, Scope: UserScope, Group: BasicGroup, EnvKey: "DISABLE_ANONYMOUS", DefaultValue: "false", ItemType: &BoolType{}, Editable: false},
```
Comment (stonezdj) [inline src/common/config/metadata/metadatalist.go:124]: To make it straightforward,  can we use AllowAnonymousUser and default value is true?

Comment (cafeliker) [reply src/common/config/metadata/metadatalist.go:124]: Do you mean allow sounds more positive than disable? ;-) I am ok with the change; and for now, the default value of DisableAnonymous is false.

Comment (steven-zou) [reply src/common/config/metadata/metadatalist.go:124]: I think we should keep consistent with the previous experience, allow anonymous. With this feature, the user can disable it via configuration.

**Comment by cafeliker** [comment]
I just did the rebase to pass DCO.

**Pull request closed without merging**